File: core/src/chrislo27/rhre/json/GameObject.java
package chrislo27.rhre.json;

public class GameObject {

	public String gameID;
	public String gameName;
	public String series;

	public SoundObject[] cues;

	public PatternObject[] patterns;

	public boolean usesGeneratorHelper = false;

	public static class SoundObject {

		public String id;
		public String fileExtension = "ogg";

		public String[] deprecatedIDs;

		public String name;

		public float duration = 0.5f;

		public boolean canAlterPitch = false;
		public boolean canAlterDuration = false;

		public String introSound = null;

		public float baseBpm = 0;

		public Boolean loops = null;
	}

	public static class PatternObject {

		public String id;
		public String name;

		public String[] deprecatedIDs;

		public boolean isStretchable = false;

		public CueObject[] cues;

		public static class CueObject {

			public String id;
			public float beat;
			public int track;

			public Float duration = 0f;

			public Integer semitone = 0;

		}
	}

}

File: core/src/chrislo27/rhre/json/GameObject.kt
package chrislo27.rhre.json

import com.fasterxml.jackson.annotation.JsonInclude

class GameObject {

	var gameID: String? = null
	var gameName: String? = null
	@JsonInclude(JsonInclude.Include.NON_DEFAULT)
	var series: String? = null

	@JsonInclude(JsonInclude.Include.NON_DEFAULT)
	var priority: Int = 0

	var cues: Array<SoundObject>? = null

	var patterns: Array<PatternObject>? = null

	@JsonInclude(JsonInclude.Include.NON_DEFAULT)
	var usesGeneratorHelper = false
	@JsonInclude(JsonInclude.Include.NON_DEFAULT)
	var notRealGame: Boolean = false

	class SoundObject {

		var id: String? = null

		@JsonInclude(JsonInclude.Include.NON_DEFAULT)
		var fileExtension = "ogg"

		@JsonInclude(JsonInclude.Include.NON_DEFAULT)
		var deprecatedIDs: Array<String>? = null

		var name: String? = null

		var duration = 0.5f

		@JsonInclude(JsonInclude.Include.NON_DEFAULT)
		var canAlterPitch = false
		@JsonInclude(JsonInclude.Include.NON_DEFAULT)
		var canAlterDuration = false

		@JsonInclude(JsonInclude.Include.NON_DEFAULT)
		var introSound: String? = null

		@JsonInclude(JsonInclude.Include.NON_DEFAULT)
		var baseBpm = 0f

		@JsonInclude(JsonInclude.Include.NON_DEFAULT)
		var pan: Float = 0f

		@JsonInclude(JsonInclude.Include.NON_DEFAULT)
		var loops: Boolean? = null
	}

	class PatternObject {

		var id: String? = null
		var name: String? = null

		@JsonInclude(JsonInclude.Include.NON_DEFAULT)
		var deprecatedIDs: Array<String>? = null

		@JsonInclude(JsonInclude.Include.NON_DEFAULT)
		var isStretchable = false

		var cues: Array<CueObject>? = null

		class CueObject {

			var id: String? = null
			var beat: Float = 0.toFloat()

			@JsonInclude(JsonInclude.Include.NON_DEFAULT)
			var track: Int = 0

			@JsonInclude(JsonInclude.Include.NON_DEFAULT)
			var duration: Float? = 0f

			@JsonInclude(JsonInclude.Include.NON_DEFAULT)
			var semitone: Int? = 0
		}
	}

}
